README.md

```diff
@@ -392,9 +395,82 @@ function determineComponent(prop) {
   if (basicType === "integer") return componentInteger(prop);
   if (basicType === "array") return componentArray(prop);
   if (basicType === "object") return componentObject(prop);
+  if (basicType === "null") return null
 
   return null;
 }
 ```
 
 Please have a look at a [reference GUI implementation here](/blob/main/reference-gui/src/gdd-gui.jsx), and [its live demo here.](https://superflytv.github.io/GraphicsDataDefinition/reference-gui/dist/)
+
+
+## Playout Options
+
+In the Schema, there is an option to add the `gddPlayoutOptions` object with various properties therein.
+These properties can be read by the playout client in order to modify how it'll display / use / play the template.
+
+_Note: All of the properties inside of `gddPlayoutOptions` are optional._
+
+```typescript
+{
+  "title": "My GFX Template",
+  "type": "object",
+  "properties": {
+    "myProperty": {
+      // ...
+    }
+  },
+  "gddPlayoutOptions": {
+    /** This object contains options that aren't tied to a specific playout server */
+    "client": {
+      /**
+       * The suggested duration of the template (in milliseconds)
+       * -1 means that it is manually taken out
+       * undefined should be treated as -1
+       * (This is ignored if steps=0)
+       * Defaults to -1
+       */
+      "duration": number,
+
+      /**
+       * Number of steps in the template
+       * 1 means that there are no steps (ie there's only "the default step").
+       * 2 or more means that it can be "stepped" (ie 2 means it can be stepped once).
+       * -1 means "infinite" number of steps.
+       * 0 means that the template is "volatile" / "fire and forget" (template really has no duration, like a bumper).
+       * Defaults to 1
+      */
+      "steps": number,
+
+      /**
+       * How the data should be formatted.
+       * This is mostly used for the older CasparCG flash-based xml data format.
+       * Defaults to "json"
+       */
+      "dataformat": "xml" | "json"
+    },
+
+    /** This object contains specific options for the various playout server types (CasparCG, Viz, vMix etc..) */
+    "playout" {
+      "casparcg": {
```

Comment:
Considering the plenitude of possible implementation combinations, should this be a part of this specification, or should we create a set of Appendixes, specifying standardized, custom payloads for specifying optional playout information. I see the value of providing a single source of information for additional "default" properties, but the sheer scope of possible implementations: AFAIK, vMix has no built in control system for HTML graphics, so it needs an OOB control channel - that alone creates an unlimited amount of possible implementations (REST API, plain-text WebSockets, Socket.IO, NodeCG shell, URL Fragments (?) etc.) for a control plane. As such, I would offload this from the main document into an Appendix, so that a claim of "GDD compatibility" is easily achievable, without the need of possibly supporting a plethora of possible control implementations.

Reply:
Good point! I'll move this into a separate document!